service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java

```diff
@@ -162,4 +166,48 @@ private void annotateDataWithQValue(List<String> studyIds, List<AlterationCountB
         return copyNumberCountByGenes;
     }
 
+    @Override
+    public List<GenericAssayDataCountItem> fetchGenericAssayDataCounts(List<String> molecularProfileIds, List<String> sampleIds,
+                                                               List<String> stableIds) throws MolecularProfileNotFoundException {
+        if (stableIds.isEmpty()) {
+            return new ArrayList<>();
+        }
+        
+        return genericAssayService.fetchGenericAssayData(molecularProfileIds, sampleIds, stableIds, "SUMMARY")
+            .stream()
+            .filter(g -> StringUtils.isNotEmpty(g.getValue()) && !g.getValue().equals("NA"))
+            .collect(Collectors.groupingBy(GenericAssayData::getGenericAssayStableId))
+            .entrySet()
+            .stream()
+            .map(entry -> {
+                int naCount = 0;
+                int totalCount = 0;
+                Map<String, List<GenericAssayData>> groupedByValue = entry.getValue().stream().collect(Collectors.groupingBy(GenericAssayData::getValue));
+                List<GenericAssayDataCount> counts = new ArrayList<>();
+                groupedByValue.forEach((value,dataList) -> {
```

Comment:
It can be replace by `map`
```
List<GenericAssayDataCount> counts = groupedByValue.entrySet().stream().map(datum -> { 
...
return dataCount;
});
```